sql/catalyst/src/main/antlr3/org/apache/spark/sql/catalyst/parser/SparkSqlLexer.g

```diff
@@ -465,7 +467,7 @@ Identifier
 fragment    
 QuotedIdentifier 
     :
-    '`'  ( '``' | ~('`') )* '`' { setText(getText().substring(1, getText().length() -1 ).replaceAll("``", "`")); }
+    '`'  ( '``' | ~('`') )* '`' { setText(getText().replaceAll("``", "`")); }
```

Comment:
Sorry I mispasted the query (github also uses backticks for escaping):

    sql("select * from (select 1 as ```weird``name`) as a")

We currently also support backticks in the name. The regex used in ```CatalystQl.cleanIdentifiers``` is quite limited and will not strip the inital and final backticks in this case. So were are expecting this:

    `weird`name

And getting this:

    ``weird`name`